chain/src/main/scala/org/bitcoins/chain/blockchain/BaseBlockChain.scala

```diff
@@ -211,4 +198,25 @@ private[blockchain] trait BaseBlockChainCompObject
 
   }
 
+  /**
+    * Finds the parent of the given header
+    */
+  private def findPrevBlockHeaderIdx(
+      header: BlockHeader,
+      blockchain: Blockchain): Option[(BlockHeaderDb, Int)] = {
+    // Let's see if we are lucky and the latest tip is the parent.
+    val prevTipOpt = blockchain.headers.headOption
+    if (prevTipOpt.map(_.hashBE).contains(header.previousBlockHashBE)) {
+      // Yes we are. Returning the latest tip.
+      prevTipOpt.map(b => (b, 0))
+    } else {
+      // No. Scanning the blockchain to find the parent.
+      // We don't want to copy the whole headers vector in zipWithIndex, so we warp it into a Stream
+      blockchain.headers.toStream.zipWithIndex.find {
```

Comment:
`.toStream()` seems to be deprecated 

```scala
  @deprecated("Use .to(LazyList) instead of .toStream", "2.13.0")
  @`inline` final def toStream: immutable.Stream[A] = to(immutable.Stream)
```

Reply:
Should we deprecate support for Scala 2.11 and 2.12 then? There's no LazyList in their standard libraries.